CMakeLists.txt

```diff
@@ -150,29 +153,30 @@ add_library(
     src/util/cable_runtime_opts_mod.F90
     src/util/masks_cbl.F90
 )
-target_link_libraries(cable_common_objlib PRIVATE PkgConfig::NETCDF)
-
-add_executable(
-    cable
-    src/offline/cable_driver.F90
-    src/offline/cable_driver_init.F90
-    "$<TARGET_OBJECTS:cable_common_objlib>"
-)
-target_link_libraries(cable PRIVATE PkgConfig::NETCDF)
-install(TARGETS cable RUNTIME)
+target_link_libraries(cable_common PRIVATE PkgConfig::NETCDF)
+if(CABLE_MPI)
+  target_compile_definitions(cable_common PRIVATE __MPI__)
+  target_link_libraries(cable_common PRIVATE MPI::MPI_Fortran)
+endif()
 
 if(CABLE_MPI)
     add_executable(
         cable-mpi
-        src/offline/cable_driver_init.F90
         src/offline/cable_mpicommon.F90
-        src/offline/cable_mpidrv.F90
         src/offline/cable_mpimaster.F90
         src/offline/cable_mpiworker.F90
         src/science/pop/pop_mpi.F90
-        "$<TARGET_OBJECTS:cable_common_objlib>"
+        src/offline/cable_offline_driver.F90
     )
-    target_compile_definitions(cable-mpi PRIVATE __MPI__)
-    target_link_libraries(cable-mpi PRIVATE PkgConfig::NETCDF MPI::MPI_Fortran)
+    target_link_libraries(cable-mpi PRIVATE cable_common MPI::MPI_Fortran)
     install(TARGETS cable-mpi RUNTIME)
+else()
+    add_executable(
+        cable
+        src/offline/cable_mpimaster_stub.F90
+        src/offline/cable_mpiworker_stub.F90
+        src/offline/cable_offline_driver.F90
+    )
+    target_link_libraries(cable PRIVATE cable_common)
+    install(TARGETS cable RUNTIME)
 endif()
```

Comment:
As discussed, this will break benchcab for the case both `cable` and `cable-mpi` executables are needed as benchcab will only invoke `cmake` once to build both executables. However benchcab will still work with these changes for serial only (`benchcab fluxsite`) or mpi only (`benchcab spatial`) tests (FYI @ccarouge).

It would be good to think about the proper CMake workflow for building different variants of the cable executable (e.g. MPI on/off, Debug, Release, compiler). [From reading online](https://cmake.org/cmake/help/latest/guide/tutorial/Packaging%20Debug%20and%20Release.html#:~:text=By%20default%2C%20CMake%27s%20model%20is%20that%20a%20build%20directory%20only%20contains%20a%20single%20configuration%2C%20be%20it%20Debug%2C%20Release%2C%20MinSizeRel%2C%20or%20RelWithDebInfo.), it seems the typical approach for doing this is to generate separate build trees for each variant unless we work with a multiconfig build system.

Reply:
> it seems the typical approach for doing this is to generate separate build trees for each variant

Yes, I think that's the usual strategy. CABLE is quite fast to compile, so this should not be an issue.

One thing you might want to explore to make this process easier is the use of CMake presets. These are really nice.

Reply:
See https://github.com/CABLE-LSM/CABLE/issues/485